### ISSUE TITLE
Autosnippets die once tab is used

### ISSUE THREAD

**Issue by curiousyogurt** [L3MON4D3/LuaSnip#901]
Autosnippets are working for me, except they cease working once I tab from the node.  I'm trying to create an autosnippet (for LaTeX) that expands a double-quotation mark (") into the standard LaTeX format for quotation marks (two backticks to open; two single quotes to close).  So, I want to type:`"Hello` and get: ```Hello''`

Here is the snippet I am currently using to do this:

```
-- local s = ls.snippet; t = ls.text_node; local i = ls.insert_node
s({ trig = '"', snippetType = "autosnippet", dscr = "Expands quotation marks into doubles" }, {
    t("``"),
    i(1),
    t("''"),
 }),
```

Everything works fine, _until_ I press tab to get out of the node (between the backticks and single quotes).  At that point, I can't use autosnippets anymore.  If I get out of the node by some other means, autosnippets continue to work.

The tab functionality (which allows me to jump out of the nodes  is handled by the following keymap (from lazyvim):

`{ "<tab>", function() require("luasnip").jump(1) end, mode = "s" },`

I'm unsure if this has to do with luasnip itself, or whether it has to do with other interactions.  Any insight would be appreciated.

**Comment by L3MON4D3** [comment]
"can't use autosnippets anymore" means they don't expand?
One thing that's a bit unusual, I guess, is that jump is only defined for "s", select-mode iirc, and not for insert as well

**Comment by curiousyogurt** [comment]
Yes, I mean they don't expand anymore once I press tab (in any context—even not when a snippet has been invoked).  And I've been playing around a bit more in trying to solve this problem, and the autosnippets stop expanding if I use a trigger to expand any snippet.  The trigger I use is set up as follows (the "¬" is option-L for me, which triggers the non-auto snippets just fine).

```
      { "¬", function()
          if require("luasnip").expand_or_jumpable() then
            require("luasnip").expand_or_jump()
          end
        end,
        mode = "i",
        desc = "LuaSnip",
      },

```

One other thing I notice, as part of the mystery, is that if I use the auto trigger as defined above (where " gets turned into ``''), then use tab to tab-out of the insert node — if I press tab again, the cursor jumps to before the text the snippet inserted—suggesting to me that it's still perhaps expecting the snippet to be completed.  I'm not really sure what's going on.

**Comment by L3MON4D3** [comment]
Huh???? xD
Okay, `if I press tab again, the cursor jumps to before the text the snippet inserted` sounds a bit like extmarks are getting messed up, could you disable any formatting/test in a filetype which certainly does not have formatting (empty filetype, `""`, for example)

But beyond that, I've no ideas other than that there's probably some conflicting plugin :/
Could you reduce your config until the issue disappears? (or, maybe start with an empty config, I think that should lead to quicker results)

**Comment by curiousyogurt** [comment]
I'll give that a try; thank you.
